doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+
+```
+  "NTP|global": {
+    "type": "hash",
+    "value": {
+      "source": "Ethernet36"
+    }
+  }
+```
+
+A change in this entry triggers hostcfgd to start the NTP configuration script, which in turn writes the ntp source to the ntp.conf and then restart the ntp service.   
+Only one global NTP source entry is allowd, and it can be either an interface name or interface IP address.   
+   
+SONiC click CLI can be extende to include this configuration.
+
+
+#### 1.1.2.3 add/delete NTP VRF
+
+This creates or deletes a global NTP vrf entry in the Redis ConfigDB. For this release, it can only be "mgmt"(??).
+
+```
+  "NTP|global": {
+    "type": "hash",
+    "value": {
+      "vrf": "mgmt"
+    }
+  }
+```
+
+A change in this DB entry triggers hostcfgd to restart ntp service.  
+    
+When Management VRF is configured, existing SONiC code always restart ntpd in the mgmt vrf context. Since this release introduces support of NTP source, 
+NTP may listen to a L3 interface in the default instance even with the presence of Management VRF. As a result, ntp service script needs to be modified 
+to restart the NTP service in the configured NTP vrf instance, or restart the ntp service in default instance if this entry is absent.   
+   
+If data vrf as NTP vrf is also supported, then hostcfgd needs to install localhost in the specified vrf in addition.
+
+SONiC click CLI can be extende to include this configuration.   
+
+
+#### 1.1.2.4 get NTP associations
+
+Transformer function issues "ntpq -p" command, parses the response and maps the outputs to the OpenConfig NTP states.
+
+### 1.1.3 Functional Requirements
+
+Provide management framework support to    
+- configure NTP server   
+- configure NTP source   
+- configure NTP vrf  
+
+### 1.1.4 Configuration and Management Requirements
+- CLI style configuration and show commands   
+- REST API support   
+- gNMI Support   
+
+Details described in Section 3.
+
+### 1.1.5 Configurations not supported by this feature using management framework:
+- NTP authenticate   
+- NTP authentication-key   
+- configure local server as a NTP server   
+- trusted key   
+- broadcast mode   
+
+### 1.1.6 Scalability Requirements
+
+### 1.1.7 Warm Boot Requirements
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+Implement NTP support using transformer in sonic-mgmt-framework.
+
+### 1.2.2 Container
+The front end code change will be done in management-framework container including:   
+- XML file for the CLI   
+- Python script to handle CLI request (actioner)   
+- Jinja template to render CLI output (renderer)   
+- OpenConfig YANG model for NTP openconfig-system.yang and openconfig-system-ext.yang   
+- SONiC NTP model for NTP based on Redis DB schema of NTP   
+- transformer functions to    
+   * convert OpenConfig YANG model to SONiC YANG model for NTP related configurations   
+   * convert from Linux command "ntpq -p" output to OpenConfig NTP state YANG model   

```

Comment:
Doesn't this need some host-side functions to send the data towards the Mgmt FW?